pkg/executorconfig/v1alpha1/executor_template_engine.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"errors"
+
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	castkey "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	cast "github.com/openebs/maya/pkg/engine"
+)
+
+type ExecutorEngine struct {
+	engine        cast.Interface // generic CAS template engine
+	defaultConfig []apis.Config  // default cas storagepool config found in CASTemplate
+	openebsConfig []apis.Config  // openebsConfig is the config that is provided
```

Comment:
can we make use of Upgrade i.e. ExecutorConfig instead of apis.Config. In other words, let us pass this Upgrade to castemplate & let it handle accordingly.
We can also think of populating the Upgrade config with exact information e.g. pools, volumes, etc before passing it to castemplate.

Reply:
will do in next pr.